### ISSUE TITLE
Live streaming: check if manifest has been updated

### ISSUE THREAD

**Issue by pavloivanchyshyn** [globocom/m3u8#116]
AFAIK, for the Live streaming, it's required to reload manifest at least after 'target_duration' for the first time and if the manifest has not changed then it must wait for a period of one-half the target duration before retrying.

So, it would be great to have an ability to check if the manifest has been already updated (e.g. using previous content).
As another possible version, it might have any sense to add a function which returns the latest line(s) from the manifest to make it possible to compare the latest line from previous manifest and current.

**Comment by leandromoreira** [comment]
I think this logic might be something to the application using the `m3u8` library, for instance, we built a monitoring tool for our live streams and there resides this logic, or re-downloading the m3u8.

**Comment by pavloivanchyshyn** [comment]
Exactly. But this logic is common. Everyone should monitor their live streams and reload them if manifest was not changed.
I guess it makes sense to add this logic to the m3u8 library to make it reusable for each user of m3u8 instead of implementing the same thing by everyone.

**Comment by leandromoreira** [comment]
Cool, so let's this open then.

**Comment by flavioribeiro** [comment]
not a commiter on this one but will give my two cents. `m3u8` was made to be a simple stateless module for HLS manifests so I wonder if maintaining update cycles or states within the module would go in the opposite direction of the module's idea. 

I'd rather create _another_ component that uses m3u8 and keep states and update cycles for live as a open source module so everyone can use it the same way they use this one.

**Comment by mradecki** [comment]
@flavioribeiro Any update on that?

**Comment by flavioribeiro** [comment]
Hey, @mradecki! my update is that I still have the same thinking about it. I haven't really signed up for any work on this, but I can see some use cases in the wild doing down to what was proposed above!
